src/snowflake/snowpark/session.py

```diff
@@ -3173,7 +3184,14 @@ def create_dataframe(
         # infer the schema based on the data
         names = None
         schema_query = None
-        if isinstance(schema, StructType):
+        if isinstance(schema, str):
+            schema = type_string_to_type_object(schema)
+            if not isinstance(schema, StructType):
+                raise ValueError(
+                    f"Invalid schema string: {schema}. "
+                    f"You should provide a valid schema string representing a struct type."
+                )
+        if isinstance(schema, (StructType, str)):
```

Comment:
would the schema still be of type `str` after being processed by the `type_string_to_type_object`?
I feel we don't need to the `str` in the instance check here
```suggestion
        if isinstance(schema, StructType):
```

Reply:
yea it's by mistake